src/QMCDrivers/tests/test_QMCCostFunctionBatched.cpp

```diff
@@ -38,4 +39,112 @@ TEST_CASE("compuate_batch_parameters", "[drivers]")
 }
 
 
+class QMCCostFunctionBatchedTest : public QMCCostFunctionBatched
+{
+public:
+  QMCCostFunctionBatchedTest(MCWalkerConfiguration& w,
```

Comment:
Please write test as functions/classes and add it as a friend to the driver class.